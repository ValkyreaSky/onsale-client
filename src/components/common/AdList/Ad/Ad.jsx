import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
// Components
import { Card, Icon, Col } from 'antd';

const { Meta } = Card;

const AdContainer = styled(Card)`
  margin-bottom: 2rem !important;
`;

const AdImageContainer = styled.div`
  height: 15rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #e8e8e8;
  color: red;
`;

const AdImage = styled.img`
  display: block;
  width: auto;
  max-width: 100%;
  height: auto;
  max-height: 15rem;
`;

const Ad = ({
  ad: {
    _id, title, image, price,
  }, withRemove, handleRemove,
}) => {
  const imageResized = image.replace(/\/upload\//, '/upload/h_150/');

  return (
    <Col
      xs={{
        span: 24,
      }}
      sm={{
        span: 12,
      }}
      md={{
        span: 6,
      }}
      lg={{
        span: 4,
      }}
    >
      <AdContainer
        hoverable
        actions={withRemove && [<Icon type="delete" onClick={handleRemove(_id)} theme="twoTone" twoToneColor="red" />, <Icon type="edit" />]}
        cover={(
          <Link to={`/ad/${_id}`}>
            <AdImageContainer>
              <AdImage data-src={imageResized} src="https://res.cloudinary.com/selli/image/upload/h_150/placeholder.png" alt={title} className="lazyload" />
            </AdImageContainer>
          </Link>
        )}
      >
        <Link to={`/ad/${_id}`}>
          <Meta
            title={title}
            description={`$${price}`}
          />
        </Link>
      </AdContainer>
    </Col>
  );
};

Ad.defaultProps = {
  withRemove: false,
  handleRemove: () => {},
};

Ad.propTypes = {
  ad: PropTypes.objectOf(PropTypes.any).isRequired,
  withRemove: PropTypes.bool,
  handleRemove: PropTypes.func,
};

export default Ad;
