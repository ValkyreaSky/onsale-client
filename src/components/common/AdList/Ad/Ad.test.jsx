import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import { Card } from 'antd';
import Ad from './Ad';

const { Meta } = Card;

describe('<Ad />', () => {
  const ad = {
    _id: 1,
    title: 'Title',
    image: 'Image',
    price: 200,
  };

  it('should match snapthot', () => {
    const wrapper = shallow(<Ad ad={ad} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Link> component with link to ad page', () => {
    const wrapper = shallow(<Ad ad={ad} />);
    expect(wrapper.find(Link).length).toBe(1);
    expect(wrapper.find(Link).props().to).toBe(`/ad/${ad._id}`);
  });

  it('should render <Meta> component with to title and price', () => {
    const wrapper = shallow(<Ad ad={ad} />);
    expect(wrapper.find(Meta).length).toBe(1);
    expect(wrapper.find(Meta).props().title).toBe(ad.title);
    expect(wrapper.find(Meta).props().description).toBe(`$${ad.price}`);
  });
});
