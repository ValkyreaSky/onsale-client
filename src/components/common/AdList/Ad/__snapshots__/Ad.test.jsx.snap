// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ad /> should match snapthot 1`] = `
<Col
  lg={
    Object {
      "span": 4,
    }
  }
  md={
    Object {
      "span": 6,
    }
  }
  sm={
    Object {
      "span": 12,
    }
  }
  xs={
    Object {
      "span": 24,
    }
  }
>
  <Styled(Card)
    actions={false}
    cover={
      <Link
        replace={false}
        to="/ad/1"
      >
        <styled.div>
          <styled.img
            alt="Title"
            className="lazyload"
            data-src="Image"
            src="https://res.cloudinary.com/selli/image/upload/h_150/placeholder.png"
          />
        </styled.div>
      </Link>
    }
    hoverable={true}
  >
    <Link
      replace={false}
      to="/ad/1"
    >
      <_default
        description="$200"
        title="Title"
      />
    </Link>
  </Styled(Card)>
</Col>
`;
